src/utils/protobuf_utils.cpp

```diff
@@ -62,6 +63,13 @@ parse_prototext_filenames_from_command_line(
     if (equal_sign != std::string::npos) {
       std::string which = s.substr(2, equal_sign-2);
       std::string fn = s.substr(equal_sign+1);
+      //check if trainer is append to filename string
+      size_t t_pos = fn.find("trainer");
+      if(t_pos != std::string::npos) {
+         //append appropriate trainer id to prototext filename
+         std::string fname = fn.substr(0,t_pos+7)+ std::to_string(trainer_rank);
+         fn = fname;
+      }
```

Comment:
I don't like how we hard-code this naming scheme where all prototext files have to end in ".trainer\<num\>". It would be better if we provided a list and trainer n could load the nth file in the list.